### PR TITLE
Log collector redux

### PR DESCRIPTION
**Pull request by BeatC**
As part of "reactifying" of non-ui logic this PR is intended to move LogCollector into new app.

### CODE REVIEW

File: react/features/base/logger/reducer.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * The log collector which captures JS console logs for this app.
+     * @type {LogCollector}
+     */
+    logCollector: null,
+
+    /**
+     * Indicates if the log collector has been started (it will not be started
+     * if the welcome page is displayed).
+     */
+    logCollectorStarted: false
```
Comment (lyubomir) [inline react/features/base/logger/reducer.js]: If the `INITIAL_STATE` is composed only of properties with values practically equal to undefined, then what's the purpose of defining them at all? Please remove them.

File: react/features/app/components/App.web.js [hunk 2]
```diff
@@ -38,6 +41,31 @@ export class App extends AbstractApp {
...
+        const store = this._getStore();
+        const {
+            logCollector,
+            logCollectorStarted
+        } = store.getState()['features/base/logger'];
+
+        window.addEventListener('beforeunload', () => {
+            // Stop the LogCollector
+            if (logCollectorStarted) {
+                logCollector.stop();
```
Comment (lyubomir) [inline react/features/app/components/App.web.js]: No, please stop piling logic into features/app and the App component. You just defined feature/base/logger so use a middleware inside it to act on actions. If that absolutely need to happen in `componentDidMount`, then introduce an `APP_DID_MOUNT` action type and middleware it inside features/base/logger.

File: react/features/app/components/App.web.js [hunk 3]
```diff
@@ -38,6 +41,31 @@ export class App extends AbstractApp {
...
+                store.dispatch(startLogCollector(false));
```
Comment (lyubomir) [inline react/features/app/components/App.web.js]: I find the function name `startLogCollector` with an argument `isStarted` that either starts or stops the log collector not a good choice. If you don't want to have `startLogCollector()` & `stopLogCollector()`, then please go with `setLogCollectorStarted(boolean)` similar to the common function name `setEnabled(boolean)`. Moreover, you already defined the redux state `logCollectorStarted` so my proposed function name sounds to me to be in line.

File: react/features/base/connection/actions.web.js [hunk 4]
```diff
@@ -31,21 +34,25 @@ export {
...
         // from the old app (at the moment of writing).
         return APP.conference.init({ roomName: room }).then(() => {
-            if (APP.logCollector) {
+            if (logCollector) {
                 // Start the LogCollector's periodic "store logs" task
-                APP.logCollector.start();
-                APP.logCollectorStarted = true;
+                logCollector.start();
+
+                dispatch(startLogCollector());
```
Comment (lyubomir) [inline react/features/base/connection/actions.web.js]: feature/base/logger implements non-UI logic and must run on both Web and React Native. Keeping in line with splitting your PR in logical, standalone commits, you may refactor Web in one commit but you must have in the PR React Native support.

**Comment by lyubomir** [comment]
@BeatC, https://github.com/jitsi/jitsi-meet/commit/7836fd199031560bf32472631ad84e4b3983a713 introduces features/base/logging so please rebase this PR on top of master. And make sure to address my previous comments, of course.

**Comment by saghul** [comment]
So much has changed it no longer makes sense to keep this open. If we need to get back to this, we can still check the closed PR.

**Pull request closed without merging**